File: leetcode-submissions/948_Bag_of_Tokens.cpp
class Solution {
public:
    int bagOfTokensScore(vector<int>& tokens, int power) {
        int score=0;
        int ans=0;
        int i=0;
        int j=tokens.size()-1;
        while(i<=j){
            if(tokens[i]<=power){
                
              
                if(ans<score){
                    ans=score;
                }
                power-=tokens[i];
                
            }else if(score>0){
                power+=tokens[j];
                score-=1;
                j--;
            }else {
                break;
            }
        }
        return ans;
    }
};

File: leetcode-submissions/948_Bag_of_Tokens.ts
function bagOfTokensScore(tokens: number[], power: number): number {
    tokens.sort((a, b)=>a-b);
    let i:number =0, j:number =tokens.length-1, maxScore:number = 0;
    let score:number =0;

    while(i<=j){
        if(power>=tokens[i]){
            power-=tokens[i];
            i++;
            score++;
            maxScore = Math.max(score , maxScore);
        }else if(score>0){
            power+=tokens[j];
            j--;
            score--;
        }else{
            break ;

        }
    }
    return maxScore ;
};